### PR TITLE
reef: mgr/cephadm: Reef Batch Backport

### PR DESCRIPTION
**Pull request by adk3798**
Backport of #49969 #50100 #50270 #50101 #50244 #50133 #50133 #50413 #50318 #50082 #50149 ~~#49953~~ #48937 #50685 #50655 #50655 #50607 #50547 #50328 #49824 #50548 #50743 #50846 #50736 #50237 #48630

Plan is to go back to normal backporting process after this, but wanted to clear up the large number of things needing a backport since reef branch creation first

Update: dropped #49953, dashboard team said this feature is not currently ready for stable branches

Update: added https://github.com/ceph/ceph/pull/50275

Update: added https://github.com/ceph/ceph/pull/50146

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nizamial09** [comment]
@adk3798 i just saw that you are also backporting https://github.com/ceph/ceph/pull/49953 which we are still delivering and not intending to backport on a stable branch yet! could you remove that from here? We are planning for a batch backport which includes all the ongoing works for multisite in the dashboard (which is still ongoing in main)

File: qa/suites/orch/cephadm/workunits/task/test_rgw_multisite.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - osd.1
+- - host.c
+  - mon.c
+  - osd.2
+tasks:
+- install:
+- cephadm:
+- cephadm.shell:
+    host.a:
+      - ceph mgr module enable rgw
```
Comment (adk3798) [inline qa/suites/orch/cephadm/workunits/task/test_rgw_multisite.yaml:18]:
```
Command failed on smithi117 with status 2: "sudo /home/ubuntu/cephtest/cephadm --image quay-quay-quay.apps.os.sepia.ceph.com/ceph-ci/ceph:352b82eda304753baf9fc4800566c992b2df518a shell -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring --fsid f757b834-d563-11ed-9aff-001a4aab830c -- bash -c 'ceph mgr module enable rgw'"
```
Think I'm missing a backport here. Most likely https://github.com/ceph/ceph/pull/50275

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
jenkins retest this please

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-04-25_18:46:10-orch:cephadm-wip-adk2-testing-2023-04-25-0929-distro-default-smithi/

Reruns of failures: https://pulpito.ceph.com/adking-2023-05-01_11:58:18-orch:cephadm-wip-adk2-testing-2023-04-25-0929-distro-default-smithi/

After reruns, 4 failures:

- 2 staggered upgrade test failures. Known issue - https://tracker.ceph.com/issues/58535
- 1 failure on test_cephfs_mirror with the agent on. These have been flaky due to the tight timeout's interaction with the agent
- 1 failure zipping logs. Saw this on some other tests run around this time. Doesn't seem like a ceph issue (failing with `stderr:grep: /home/ubuntu/cephtest/archive/syslog/kern.log: Permission denied`)

Nothing there to block merging

**Review by asm0deuz — APPROVED** [review]
LGTM

**Comment by adk3798** [comment]
jenkins retest this please

**Pull request merged**